chore : Testing travis CI #15

RyuJaesang opened the pull request:
Travis CI 도큐먼트대로 .travis.yml 작성하였습니다.
Travis CI에 대한 허용만 오너(@hea9549 @hihiboss )님들이 허용 부탁드려요~
그리고 .travis.yml은 그냥 공식 도큐 설정을 그대로 가져왔으므로 수정 사항 필요하면 말씀 부탁드립니다.
resolved : #15 
resolves: #19

hihiboss commented on PULL_REQUEST_TEMPLATE/feature-template.md:
```diff
@@ -1,13 +1,13 @@
+## Related Issue
+Create issue and tag it
+
+## Checklist
```
여기 체크리스트들 대괄호 사이에 스페이스 넣어야 체크박스로 표시됩니다!

RyuJaesang replied:
엌ㅋㅋㅋㅋ 죄송함당

hihiboss commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+  - "3.4"
+  - "3.5"
+  - "3.6"      # current default Python on Travis CI
+  - "3.7"
```
저희 파이썬 버전 3.7 쓰니까 3.7만 기재해주세요~

hihiboss commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+  - pip install -r requirements.txt
+# command to run tests
+script:
+  - pytest
```
저희 `requirements.txt`에 pytest 의존성 없는데 스크립트 동작하나요?! 
제 로컬환경에서는 동작안해서.. 확인부탁드립니다~

RyuJaesang replied:
앞서 언급하였듯이 현재 프로젝트에 travis에 대한 허용을 제가 할 수 없어서 이 환경에선 테스트 해보지 못했습니다

hihiboss replied:
https://github.com/RyuJaesang/my-community-backend 에서 성공하면 됩니다~

hihiboss commented:
추가로 현재 커밋한 author가 알수없음(깃헙 아이콘이 회색고양이임)인데 자기 커밋 관리해야하면 이 부분도 확인해주세요~

RyuJaesang commented:
내일 퇴근하고 전부 혼쭐내놓겠습니다~
빠른 리뷰 감사합니다~!

Commit:
chore : Testing travis CI

chore : Testing travis CI

chore: Resolve Issue #25

chore: Add pytest and pytest-django dependency

feat: junhee remote pull

chore: fix run script

fix: travis raise error in the event of no log until 10min (add trash log)

feat: add test code for travis test build

fix: PR template

feat: conversation resolve

chore: delete untracked file

chore : rebase merge

RyuJaesang commented:
PR template : #25 해-결
Tavis CI : 준희가 올린 PR 보니까 pytest에 의존성 넣었던데 그거 머지되고 추가적으로 테스팅하면서 진행해볼게

GunHee5719 commented on .travis.yml:
```diff
@@ -0,0 +1,10 @@
+  - pip install -r requirements.txt
+# command to run tests
+script:
+  - pytest
```
아 그리고 혹시, 찾아보니까 travis 적용 시 이 부분에서 pyLint를 적용이 가능한 것 같은데 이 부분 확인해주실 수 있을까요~?

RyuJaesang replied:
네~ 지금부터 진행해볼게요~

RyuJaesang commented:
Travis CI test를 위해 준희가 올린 pytest.ini 양식에 맞춰 dummy_test.py를 추가하여 테스트 하였고, 해결했습니다~
확인하고 approve 부탁드립니다:)

RyuJaesang commented:
Tracis ci의 테스트를 위해 @hihiboss 가 올린 pytest.ini 형식에 맞춰 dummy_test.py로 테스트를 진행하여 연동하였습니다.(추후 삭제 필요)
또한 @GunHee5719 가 언급한 travis내에서의 lint 설정은 검색해도 나오지 않아서.. 혹시 관련 링크 있으면 알려주시면 감사하겠습니다~!

hihiboss commented:
@RyuJaesang @GunHee5719 

```
$ pytest --pylint
```
.travis.yml의 scripts를 위처럼 변경하거나

```
$ pylint conf/*
```
위처럼 추가하면 될 것 같습니다~

hihiboss commented:
커밋이 너무 많아서 #14 머지 후에 스쿼시로 머지해야 할 것 같습니다!

hihiboss reviewed (changes requested):
커밋이 너무 많아서 #14 머지 후에 스쿼시로 머지해야 할 것 같습니다.

#14 머지 전까지 대기!!

RyuJaesang commented:
네 알겠습니다 내일까지 린트 컨피그 파일까지 대충 작성하겠습니다:)

hihiboss commented:
@RyuJaesang 

그리고 해당 PR이 머지되어 있는 RyuJaesang/my-community-backend 에서 확인해보니
![image](https://user-images.githubusercontent.com/34789831/87143911-e003d880-c2e1-11ea-963b-bf04c65c3c59.png)
이처럼 PR 템플릿이 반영 안되네요 ㅜㅜ
제 부족한 영어 때문에 못보신거 같은데 PR 템플릿은 `PULL_REQUEST_TEMPLATE.md`라는 파일 하나로만 설정 가능한 걸로 알고 있습니다!

RyuJaesang commented:
수정했습니다. 제가 어제 브랜치 전부 병합해서 풀리퀘를 테스트해볼만한 브랜치가 없는데 확인 부탁드릴게요:):):):):):):)

hihiboss commented:
![image](https://user-images.githubusercontent.com/34789831/87239077-cd0a1900-c445-11ea-826b-c30939a44331.png)

@RyuJaesang 

PR템플릿 적용됩니다~!

hihiboss reviewed (changes requested):
1. develop 브랜치 rebase 받고 conflict 해결하기 (requirements.txt)
2. travis에 lint 적용하기

GunHee5719 commented on .travis.yml:
```diff
@@ -0,0 +1,12 @@
+  - travis_wait 1 pip install -r requirements.txt
+# command to run tests
+script:
+  - python manage.py runserver
```
혹시 pyLint는 적용 안되던가요..? 안되면 우선은 충돌 해결됐으니 travis 머지하는게 좋을거 같습니다.

hihiboss replied:
**request**: 서버 실행 말고 테스트 코드를 실행해야 합니다!

hihiboss reviewed (changes requested):
**request**: myvenv 관련 파일 모두 ignore 해주세요.

hihiboss commented:
**suggestion**: 커밋 보니까 꼬이고 난리난거 같아서 develop에 upstream develop 리베이스 받고 새로 브랜치 만들어서 작업하기를 추천합니다.

RyuJaesang commented:
Upstream develop rebase하고 제꺼 머지시켰습니다.
커밋도 전부 리베이스해서 1개만 남겼습니다.
고인 형아들이 보시고 제대로 됐는지 확인부탁드려요 👍 
@hihiboss @GunHee5719 @hea9549 @als95 @jhcheon1000

RyuJaesang committed:
chore : pylint exit 0 for pass TravisCI

RyuJaesang commented:
![image](https://user-images.githubusercontent.com/28643516/89304570-8b693880-d6a8-11ea-8bc3-bb3f06ffdaa5.png)
그리고 현재 위와 같이 Pylint를 적용했을 때 exit 28이 출력되는데 이는 코드 퀄리티가 부족해서 출력되는 non-zero exit code라
이를 임시로 해결하기위해 exit 0를 강제하였습니다.

hihiboss commented:
나중에 lint 관련 설정 건드려서 기준 점수를 낮춰야할듯 ;;
수고했습니당~

hea9549 commented:
LGTM :)

hea9549 commented:
다른이견 없으면 8월 8일 24시 이전/후로 머지합니다

GunHee5719 reviewed (approved):
Good:)
